extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/WiringProcessor.java

```diff
@@ -0,0 +1,243 @@
+                    if (isOutboundConnector(bi.getImplClazz())) {
+                        builder.produce(
+                                ConnectorBuildItem.createOutboundConnector(getConnectorName(bi),
+                                        getConnectorAttributes(bi, index,
+                                                ConnectorAttribute.Direction.OUTGOING,
+                                                ConnectorAttribute.Direction.INCOMING_AND_OUTGOING)));
+                    }
+                });
+    }
+
+    @BuildStep
+    public void collectAllApplicationChannels(BuildProducer<IncomingChannelBuildItem> incomings,
+            BuildProducer<OutgoingChannelBuildItem> outgoings,
+            CombinedIndexBuildItem index) {
+
+        Collection<AnnotationInstance> annotations = index.getIndex().getAnnotations(ReactiveMessagingDotNames.INCOMING);
+        for (AnnotationInstance annotation : annotations) {
+            String name = annotation.value().asString();
+            Optional<String> managingConnector = getManagingConnector("incoming", name);
+            incomings.produce(IncomingChannelBuildItem.of(name, managingConnector.orElse(null)));
+        }
+
+        annotations = index.getIndex().getAnnotations(ReactiveMessagingDotNames.INCOMINGS);
+        for (AnnotationInstance annotation : annotations) {
+            for (AnnotationValue channel : annotation.values()) {
+                String name = channel.asString();
+                Optional<String> managingConnector = getManagingConnector("incoming", name);
+                incomings.produce(IncomingChannelBuildItem.of(name, managingConnector.orElse(null)));
+            }
+        }
+        annotations = index.getIndex().getAnnotations(ReactiveMessagingDotNames.OUTGOING);
+        for (AnnotationInstance annotation : annotations) {
+            String name = annotation.value().asString();
+            Optional<String> managingConnector = getManagingConnector("outgoing", name);
+            outgoings.produce(OutgoingChannelBuildItem.of(name, managingConnector.orElse(null)));
+        }
+
+        annotations = index.getIndex().getAnnotations(ReactiveMessagingDotNames.CHANNEL);
+        for (AnnotationInstance annotation : annotations) {
+            String name = annotation.value().asString();
+            if (!name.isBlank()) {
+                if (isEmitter(annotation)) {
+                    Optional<String> managingConnector = getManagingConnector("outgoing", name);
+                    outgoings.produce(OutgoingChannelBuildItem.of(name, managingConnector.orElse(null)));
+                } else {
+                    Optional<String> managingConnector = getManagingConnector("incoming", name);
+                    incomings.produce(IncomingChannelBuildItem.of(name, managingConnector.orElse(null)));
+                }
+            }
+        }
+    }
+
+    @BuildStep
+    public void detectOrphanChannels(List<IncomingChannelBuildItem> incomings,
```

Comment:
I think we need to merge this with the other processor. The `SmallRyeReactiveMessagingProcessor` captured other mediator components except for Connectors and connector managed channels. Now that we have those, we can run the wiring logic (currently running at runtime) at build time, and provide the wiring result to the runtime.

Reply:
I've merged everything in the `WiringProcessor`.